lxd/daemon.go

```diff
@@ -641,84 +638,25 @@ func (d *Daemon) Authenticate(w http.ResponseWriter, r *http.Request) (*request.
 	return &request.RequestorArgs{Trusted: false}, nil
 }
 
-// handleOIDCAuthenticationResult checks the identity cache for the OIDC identity by their email address. If no identity
-// is found, an identity is added with that email. If an identity is found but the OIDC subject is different to the
-// expected value, the identity is updated with the new subject.
-func (d *Daemon) handleOIDCAuthenticationResult(r *http.Request, result *oidc.AuthenticationResult) error {
-	var action lifecycle.IdentityAction
-
-	id, err := d.identityCache.Get(api.AuthenticationMethodOIDC, result.Email)
-	if err != nil && !api.StatusErrorCheck(err, http.StatusNotFound) {
-		return fmt.Errorf("Failed getting OIDC identity from cache: %w", err)
-	} else if err != nil {
-		// Identity not found. Add it to the database and refresh the identity cache.
-		idMetadata := dbCluster.OIDCMetadata{Subject: result.Subject}
-		b, err := json.Marshal(idMetadata)
-		if err != nil {
-			return fmt.Errorf("Failed to marshal OIDC identity metadata: %w", err)
-		}
-
-		err = d.db.Cluster.Transaction(d.shutdownCtx, func(ctx context.Context, tx *db.ClusterTx) error {
-			_, err := dbCluster.CreateIdentity(ctx, tx.Tx(), dbCluster.Identity{
-				AuthMethod: api.AuthenticationMethodOIDC,
-				Type:       api.IdentityTypeOIDCClient,
-				Identifier: result.Email,
-				Name:       result.Name,
-				Metadata:   string(b),
-			})
-			return err
-		})
-		if err != nil {
-			return fmt.Errorf("Failed to add new OIDC identity to database: %w", err)
-		}
-
-		action = lifecycle.IdentityCreated
-	} else if id.Subject != result.Subject || id.Name != result.Name {
-		// The OIDC subject of the user with this email address has changed (this should be rare). Replace the
-		// subject in the identity metadata and refresh the cache.
-		idMetadata := dbCluster.OIDCMetadata{Subject: result.Subject}
-		b, err := json.Marshal(idMetadata)
-		if err != nil {
-			return fmt.Errorf("Failed to marshal OIDC identity metadata: %w", err)
-		}
-
-		err = d.db.Cluster.Transaction(d.shutdownCtx, func(ctx context.Context, tx *db.ClusterTx) error {
-			return dbCluster.UpdateIdentity(ctx, tx.Tx(), api.AuthenticationMethodOIDC, result.Email, dbCluster.Identity{
-				AuthMethod: api.AuthenticationMethodOIDC,
-				Type:       api.IdentityTypeOIDCClient,
-				Identifier: result.Email,
-				Name:       result.Name,
-				Metadata:   string(b),
-			})
-		})
-		if err != nil {
-			return fmt.Errorf("Failed to update OIDC identity information: %w", err)
-		}
-
-		action = lifecycle.IdentityUpdated
+// handleOIDCAuthenticationResult checks the identity cache for the OIDC identity by their email address.
+// If no identity is found, the cache is refreshed.
+// This is for new OIDC logins and is currently required for authorization to work because group membership is saved to the cache.
```

Comment:
and if group membership changes does this trigger a cluster notification to refresh the cache?